app/upgrades/v20/upgrades.go

```diff
@@ -25,7 +46,80 @@ func CreateUpgradeHandler(
 			return vm, err
 		}
 
+		InitializeMaxProviderConsensusParam(ctx, keepers.ProviderKeeper)
+		InitializeMissingValidatorSetCaps(ctx, keepers.ProviderKeeper)
+		SetMaxValidators(ctx, *keepers.StakingKeeper)
+		InitializeLastProviderConsensusValidatorSet(ctx, keepers.ProviderKeeper, *keepers.StakingKeeper)
+
 		ctx.Logger().Info("Upgrade v20 complete")
 		return vm, nil
 	}
 }
+
+// InitializeMissingValidatorSetCaps initializes the max validators
+// parameter for existing consumers to the MaxProviderConsensusValidators parameter.
+// This is necessary to avoid those consumer chains having an excessive amount of validators.
+func InitializeMissingValidatorSetCaps(ctx sdk.Context, providerKeeper providerkeeper.Keeper) {
+	for _, chainID := range providerKeeper.GetAllRegisteredConsumerChainIDs(ctx) {
+		providerKeeper.SetValidatorSetCap(ctx, chainID, NewMaxProviderConsensusValidators)
+	}
+}
+
+// InitializeMaxProviderConsensusParam initializes the MaxProviderConsensusValidators parameter.
+// It is set to 180, which is the current number of validators participating in consensus on the Cosmos Hub.
+// This parameter will be used to govern the number of validators participating in consensus on the Cosmos Hub,
+// and takes over this role from the MaxValidators parameter in the staking module.
+func InitializeMaxProviderConsensusParam(ctx sdk.Context, providerKeeper providerkeeper.Keeper) {
+	params := providerKeeper.GetParams(ctx)
+	if params.MaxProviderConsensusValidators == 0 {
+		params.MaxProviderConsensusValidators = NewMaxProviderConsensusValidators
+		providerKeeper.SetParams(ctx, params)
+	}
+}
+
+// SetMaxValidators sets the MaxValidators parameter in the staking module to 200,
+// which is the current number of 180 plus 20.
+// This is done in concert with the introduction of the inactive-validators feature
+// in Interchain Security, after which the number of validators
+// participating in consensus on the Cosmos Hub will be governed by the
+// MaxProviderConsensusValidators parameter in the provider module.
+func SetMaxValidators(ctx sdk.Context, stakingKeeper stakingkeeper.Keeper) {
+	params, err := stakingKeeper.GetParams(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	params.MaxValidators = NewMaxValidators
+
+	err = stakingKeeper.SetParams(ctx, params)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// InitializeLastProviderConsensusValidatorSet initializes the last provider consensus validator set
+// by setting it to the first 180 validators from the current validator set of the staking module.
+func InitializeLastProviderConsensusValidatorSet(ctx sdk.Context, providerKeeper providerkeeper.Keeper, stakingKeeper stakingkeeper.Keeper) {
```

Comment:
Thanks @insumity, yes the `LastProviderConsensusValSet` should be initialized here. Nice catch